web/massastation/src/custom/api/usePost.ts

```diff
@@ -8,15 +8,28 @@ import { UseMutationResult, useMutation } from '@tanstack/react-query';
 
 export function usePost<T, P = unknown>(
   resource: string,
-): UseMutationResult<P, unknown, T, unknown> {
-  var url = `${import.meta.env.VITE_BASE_API}/${resource}`;
+): UseMutationResult<
+  P,
+  unknown,
+  { params?: Record<string, string>; payload?: T },
```

Comment:
can you explain this?

Reply:
these are the arguments that you give when we call the mutation,
ex
```ts
    const payload = { name: 'John Doe', age: 25 };
    const params = { token: 'abc123' };

    mutate({  payload, params });
```